lib/matplotlib/ticker.py

```diff
@@ -511,53 +586,97 @@ def format_data(self, value):
         s = self._formatSciNotation(s)
         return self.fix_minus(s)
 
+    def set_offset_string(self, value):
+        """
+        Set the string which typically contains the offset
+        or the scaling factor which is to be displayed on the axis.
+        Set this to None to  allow the string set by offset or scaling
+        algorithm.
+
+        Parameters
+        ----------
+        s:  String describing the offset
+
+        Returns
+        -------
+        None
+        """
+        self._offsetString = value
+
     def get_offset(self):
-        """Return scientific notation, plus offset"""
+        """
+        Returns a string with the offset(or scientific notation)/scaling
+        factor which is properly formatted. This is used as additional text
+        next to the ticks, either determined by offset/scaling or set by the
+        user
+
+        Parameters
+        ----------
+        NONE
+
+        Returns
+        -------
+        :string
+        """
+        # String has been set manually, so just return that
+        if self._offsetString is not None:
+            return self._offsetString
+
         if len(self.locs) == 0:
             return ''
         s = ''
-        if self.orderOfMagnitude or self.offset:
-            offsetStr = ''
-            sciNotStr = ''
-            if self.offset:
-                offsetStr = self.format_data(self.offset)
-                if self.offset > 0:
-                    offsetStr = '+' + offsetStr
-            if self.orderOfMagnitude:
-                if self._usetex or self._useMathText:
-                    sciNotStr = self.format_data(10 ** self.orderOfMagnitude)
-                else:
-                    sciNotStr = '1e%d' % self.orderOfMagnitude
-            if self._useMathText:
-                if sciNotStr != '':
-                    sciNotStr = r'\times%s' % _mathdefault(sciNotStr)
-                s = ''.join(('$', sciNotStr, _mathdefault(offsetStr), '$'))
-            elif self._usetex:
-                if sciNotStr != '':
-                    sciNotStr = r'\times%s' % sciNotStr
-                s = ''.join(('$', sciNotStr, offsetStr, '$'))
+        offsetStr = ''
+        sciNotStr = ''
+        if self.get_useOffset() is True:
```

Comment:
I am assuming that there is a latch someplace that sets this to False if `self.offsetval == 0`?